plugins/by-name/yaml-companion/default.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  config,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "yaml-companion";
+  packPathName = "yaml-companion.nvim";
+  package = "yaml-companion-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  settingsExample = {
+    schemas = [
+      {
+        name = "Argo CD Application";
+        uri = "https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/argoproj.io/application_v1alpha1.json";
+      }
+      {
+        name = "SealedSecret";
+        uri = "https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/bitnami.com/sealedsecret_v1alpha1.json";
+      }
+    ];
+    lspconfig = {
+      settings = {
+        yaml = {
+          format.enable = false;
+          schemaStore = {
+            enable = false;
+            url = "";
+          };
+          schemas = lib.nixvim.nestedLiteralLua "require('schemastore').yaml.schemas()";
+        };
+      };
+    };
+  };
+
+  callSetup = false;
+  extraConfig = cfg: {
+    warnings = lib.nixvim.mkWarnings "plugins.yaml-companion" [
+      {
+        when = !config.plugins.lsp.enable;
```

Comment:
This should support either `lsp` or `plugins.lsp`.

See example: https://github.com/nix-community/nixvim/blob/2c0a9ff1e2bcc6aab15caa504a7c9670f6e0a929/plugins/by-name/schemastore/default.nix#L132

Note that `lsp` doesn't have an enable option, so we'd want to check the relevant server is enabled.

For `plugins.lsp`, we may want to `mkDefault` the relevant server. This warning may want to check the `plugins.lsp.servers` enable option too?